test/name-mapping.test.js

```diff
@@ -0,0 +1,135 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: loopback-connector
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const expect = require('chai').expect;
+const Connector = require('../lib/connector');
+const testConnector = require('./connectors/test-connector');
+
+const juggler = require('loopback-datasource-juggler');
+const ds = new juggler.DataSource({
+  connector: testConnector,
+  debug: true,
+});
+
+/* eslint-disable one-var */
+let connector;
+let Customer;
+let Order;
+/* eslint-enable one-var */
+
+describe('Name mapping', function() {
+  let connector, builder;
+
+  before(() => {
+    connector = ds.connector;
+    connector._tables = {};
+    connector._models = {};
+    Customer = ds.createModel(
+      'Customer',
+      {
+        name: {
+          id: true,
+          type: String,
+          testdb: {
+            column: 'FIRSTNAME',
+            dataType: 'VARCHAR',
+            dataLength: 32,
+          },
+        },
+        middleName: {
+          type: Boolean,
+          name: 'middle_name',
+          postgresql: {
+            column: 'MIDDLENAME',
+          },
+        },
+        lastName: {
+          type: Boolean,
+          testdb: {
+            column: 'LASTNAME',
+          },
+        },
+        primaryAddress: {
+          type: String,
+          name: 'primary_address',
+        },
+        address: String,
+      },
+      {testdb: {table: 'CUSTOMER'}},
+    );
+
+    // use field in this model to mock NoSQL DB
+    Order = ds.createModel(
+      'Order',
+      {
+        id: {
+          id: true,
+          type: Number,
+          testdb: {
+            field: 'my_id',
+            dataType: 'ObjectId',
+          },
+        },
+        des: {
+          type: String,
+          testdb: {
+            field: 'DESCRIPTION',
+          },
+        },
+      },
+      {testdb: {table: 'ORDER'}},
+    );
+  });
+
+  context('idMapping', function() {
+    it('should map id column name', function() {
+      const idCol = connector.idMapping('Customer');
+      expect(idCol).to.eql('FIRSTNAME');
+    });
+
+    it('alias idColumn should map id column name', function() {
+      const idCol = connector.idColumn('Customer');
+      expect(idCol).to.eql('FIRSTNAME');
+    });
+
+    it('should map id field name', function() {
+      const idCol = connector.idMapping('Order');
+      expect(idCol).to.eql('my_id');
+    });
+  });
+  context('propertyMapping', function() {
+    it('prefers property name when database is different', function() {
```

Comment:
The test case here is meant to test that when a provided database name is different the current connector name, the setting would get ignored. So in this test, `postgresql.xxx` gets ignored because it was expecting the name to be `testdb.xxxx`. The result would be the same if we use other names such as `mongodb`.

Reply:
oh I misunderstood the purpose of this test. 👍 fair enough

Reply:
Let me reword the title